### PR TITLE
trealla: only put valgrind in checkInputs if not darwin

### PR DESCRIPTION
**Pull request by siraben**
###### Description of changes
Fixes breakage caused by #215567

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 219568` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trealla</li>
  </ul>
</details>

File: pkgs/development/interpreters/trealla/default.nix [hunk 1]
```diff
@@ -27,7 +27,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ xxd ];
   buildInputs = [ readline openssl libffi ];
-  checkInputs = [ valgrind ];
+  checkInputs = lib.optionals (!stdenv.isDarwin) [ valgrind ];
```
Comment (GabrielDougherty) [inline pkgs/development/interpreters/trealla/default.nix:30]: Valgrind builds still work on [x86_64-darwin](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.valgrind.x86_64-darwin) so this should instead be:

```suggestion
  checkInputs = lib.optionals (!(stdenv.isDarwin && stdenv.isAarch64)) [ valgrind ];
```

**Review by GabrielDougherty — CHANGES_REQUESTED** [review]

**Comment by GabrielDougherty** [comment]
The diff LGTM. I suggest rebasing this branch to combine both commits in this PR into one commit that has the message

```
trealla: only put valgrind in checkInputs if not aarch64-darwin

because valgrind does not work on Apple Silicon
```

In line with https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages

**Review by GabrielDougherty — APPROVED** [review]

**Comment by siraben** [comment]
Ok, will rebase into one commit.

**Review by thoughtpolice — APPROVED** [review]
sorry about that; lgtm

**Pull request merged**